### PR TITLE
Add json formatter and change how loggers are retrieved

### PR DESCRIPTION
**Pull request by ohayak**
I added an option to logged to output logs in json format with possibility to add customized keys.
The second change is about the behavior when @logged(mylogger) is used.
Let's take this example:
```
mylogger = logging.getLogger('mylogger')
@logged(mylogger)
class Examle:
...
```
Im expecting from logged to use mylogger, and a call to __log will be behave exactly like a call to mylogger.
but after executing this these lines in logged function:
```
if isinstance(obj, logging.Logger): # `@logged(logger)'
        return lambda class_or_fn: _add_logger_to(class_or_fn,
            logger_name=_generate_logger_name(class_or_fn, parent_name=obj.name))
```
logged will use a different logger under the name 'mylogger.Example'
in other words logged always creates a new logger.
I changed this behaviour to retrieve logger name and use it instead. and create new logger only if no logger were given.

### CODE REVIEW

**Comment by mzipay** [comment]
Hi @ohayak,

I'm not going to merge this PR into the mainline, but not because I disagree with your ideas - they're both good ideas, particularly the ability to emit JSON-formatted log entries.

One of the primary design goals of Autologging is to **avoid** baking in capabilities that the standard library's logging module provides. Creating a logger that can emit JSON-formatted entries is relatively easy to do, and Autologging will happily delegate to such a logger.  So this is not something I'd include in Autologging itself.

On the matter of the log naming - what you observe is by design, and documented clearly as such.  When you provide a logger object, it's name is taken as the _parent_ logger, and the class's name is appended to that in order to derive the log channel for that class's __log member.  This behavior is absolutely intentional, and not a behavior I want to change.
However, the desire to specify a log channel and _not_ have Autologging append the class name is a perfectly valid use case.  This will be possible in the next version of Autologging. Using your earlier example, you will simply use `@logged("mylogger")` (note the argument is a string here) and Autologging will use that channel as-is.
